File: src/Python/Parcial2/ActividadP2A4/empleadoasalariado.py
from empleado import Empleado

class EmpleadoAsalariado(Empleado):
    def __init__(self, nombre, apellido, nss, salario):
        super().__init__(nombre, apellido, nss)
        self.establecerSalarioSemanal(salario)

    def establecerSalarioSemanal(self, salario):
        if salario < 0.0:
            self.salarioSemanal = 0.0
        else:
            self.salarioSemanal = salario

    def obtenerSalarioSemanal(self):
        return self.salarioSemanal

    def ingresos(self):
        return self.obtenerSalarioSemanal()

    def __str__(self):
        return f"empleado asalariado: {super().__str__()}\nsalario semanal: ${self.obtenerSalarioSemanal()}"

File: src/C#/Parcial2/ActividadP2A4/EmpleadoAsalariado.cs
using System;

public class EmpleadoAsalariado : Empleado
{
    private double salarioSemanal;

    public EmpleadoAsalariado(string nombre, string apellido, string nss, double salario)
        : base(nombre, apellido, nss)
    {
        EstablecerSalarioSemanal(salario);
    }

    public void EstablecerSalarioSemanal(double salario)
    {
        salarioSemanal = salario < 0.0 ? 0.0 : salario;
    }

    public double ObtenerSalarioSemanal() { return salarioSemanal; }

    public override double Ingresos() { return ObtenerSalarioSemanal(); }

    public override string ToString()
    {
        return "empleado asalariado: " + base.ToString() + "\nsalario semanal: $" + ObtenerSalarioSemanal();
    }
}
